modules/tinymce/src/core/main/ts/content/EditorContent.ts

```diff
@@ -5,14 +5,54 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Cell } from '@ephox/katamari';
+
+import { addDefaultGetFormats } from 'tinymce/core/content/GetContentImpl';
+import { addDefaultSetFormats } from 'tinymce/core/content/SetContentImpl';
+
+import Editor from '../api/Editor';
+import * as AddFormat from './AddFormat';
 import { Content, GetContentArgs, SetContentArgs } from './ContentTypes';
 import { getContent } from './GetContent';
 import { setContent } from './SetContent';
 
+interface EditorContent {
+  addFormat: (format: string, formatGetter: GetContentFormatter, formatSetter: SetContentFormatter) => void;
+  get: (args: Partial<GetContentArgs>) => Content;
+  set: (content: Content, args?: Partial<SetContentArgs>) => Content;
+}
+
+export type GetContentFormatter = (editor: Editor, args: GetContentArgs) => Content;
+export type SetContentFormatter = (editor: Editor, content: Content, args: SetContentArgs) => Content;
+
+const EditorContent = (editor: Editor): EditorContent => {
+  const setCell: Cell<Record<string, SetContentFormatter>> = Cell<Record<string, SetContentFormatter>>({});
+  const getCell: Cell<Record<string, GetContentFormatter>> = Cell<Record<string, GetContentFormatter>>({});
```

Comment:
I'm not really a fan of these names that are used everywhere, as it's not entirely clear. Can we rename them to something like `getContentRegistry` and `setContentRegistry`?

Also we probably shouldn't be passing the cells around, as the addition of new formats should only happen in this file (see how `Mode.ts` does this). As an example, instead of passing `getCell` directly below for `getContent`, we'd update it to do `getCell()` and pass the record around instead

- [ ] Don't pass the cells every, only the "registry"

Reply:
We strictly don't need the cell anywhere, as the value it contains ( the object ) is always the same. I was just under the impression that we want to toss around cells instead. If that isn't what we want to do, I can change it, though I'd like to make sure I've understood it correctly: You want to remove the `Cell<>`from around the record where possible?

Reply:
Yes that's correct. We should only be passing the cell where the "registry" can be modified. That should not be happening in the get/set content case as it should only be reading from the "registry", so we should only be passing the cell in the add format case.